trino-aws-proxy-spi/src/main/java/io/trino/aws/proxy/spi/signing/RequestAuthorization.java

```diff
@@ -26,22 +27,34 @@
 import static com.google.common.collect.ImmutableSet.toImmutableSet;
 import static java.util.Objects.requireNonNull;
 
-public record RequestAuthorization(String authorization, String accessKey, String region, String keyPath, Set<String> lowercaseSignedHeaders, String signature, Optional<String> securityToken)
+public record RequestAuthorization(String accessKey, String region, String keyPath, Set<String> lowercaseSignedHeaders, String signature,
+                                   Optional<Instant> expiry, Optional<String> securityToken)
 {
+    public static final RequestAuthorization INVALID = new RequestAuthorization("", "", "", ImmutableSet.of(), "", Optional.empty(), Optional.empty());
+    private static final String SIGNATURE_ALGORITHM = "AWS4-HMAC-SHA256";
+
     public RequestAuthorization
     {
-        requireNonNull(authorization, "authorization is null");
         requireNonNull(accessKey, "accessKey is null");
         requireNonNull(region, "region is null");
         requireNonNull(keyPath, "keyPath is null");
         lowercaseSignedHeaders = ImmutableSet.copyOf(lowercaseSignedHeaders);
         requireNonNull(signature, "signature is null");
+        requireNonNull(expiry, "expiry is null");
         requireNonNull(securityToken, "securityToken is null");
     }
 
     public boolean isValid()
     {
-        return !lowercaseSignedHeaders.isEmpty() && !signature.isEmpty() && !accessKey.isEmpty() && !region.isEmpty();
+        return !lowercaseSignedHeaders.isEmpty() && !signature.isEmpty() && !accessKey.isEmpty() && !region.isEmpty() && expiry.map(expiryTimestamp -> Instant.now().isBefore(expiryTimestamp)).orElse(true);
+    }
+
+    public String authorization()
+    {
+        if (expiry.isPresent()) {
```

Comment:
Please use Preconditions. e.g. `checkState(expiry.isEmpty(), "authorization cannot be computed for an expiring request");`